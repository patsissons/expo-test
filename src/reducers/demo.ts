import { Action } from 'redux';
import { actionTypes } from '../navigation/actions';

const initialState = {
  count: 0,
}

export function reducer(state = initialState, action: Action) {
  switch (action.type) {
    case actionTypes.demo.Increment:
      return {
        count: state.count + 1,
      };
  }

  return state;
}
