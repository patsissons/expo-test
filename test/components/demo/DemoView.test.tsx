import React from 'react';
import { create } from 'react-test-renderer';
import { DemoView } from '../../../src/components/demo/DemoView';

describe('DemoView', () => {
  it('renders without crashing', () => {
    // Arrange
    const component = (<DemoView />);

    // Act
    const result = create(component).toJSON();

    // Assert
    expect(result).toMatchSnapshot();
  });

  it('supports injecting custom initial count', () => {
    // Arrange
    const initialCount = 123;
    const app = (<DemoView initialCount={ initialCount } />);

    // Act
    const result = create(app).root;
    const state = result.findByProps({ testID: 'state' }).props.children;

    // Assert
    expect(result.props).toEqual({ initialCount });
    expect(state).toMatch(`${ initialCount }`);
  });
});
