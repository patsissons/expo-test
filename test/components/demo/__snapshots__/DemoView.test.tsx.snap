// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DemoView renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DemoComponent
    actions={
      Object {
        "demo": Object {
          "increment": [Function],
        },
        "navigation": Object {
          "goBack": [Function],
          "toLogin": [Function],
          "toMain": [Function],
        },
      }
    }
    header="Current State is "
    state={
      Object {
        "count": 0,
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          testID="state"
        >
          Current State is 0
        </Text>,
        <Button
          onPress={[Function]}
          title="increment"
        >
          Increment Count
        </Button>,
      ],
      "style": Array [
        Object {},
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "flex": 1,
          "justifyContent": "center",
        },
        Object {},
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "Current State is 0",
          "ellipsizeMode": "tail",
          "testID": "state",
        },
        "ref": null,
        "rendered": "Current State is 0",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Increment Count",
          "onPress": [Function],
          "title": "increment",
        },
        "ref": null,
        "rendered": "Increment Count",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            testID="state"
          >
            Current State is 0
          </Text>,
          <Button
            onPress={[Function]}
            title="increment"
          >
            Increment Count
          </Button>,
        ],
        "style": Array [
          Object {},
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "flex": 1,
            "justifyContent": "center",
          },
          Object {},
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Current State is 0",
            "ellipsizeMode": "tail",
            "testID": "state",
          },
          "ref": null,
          "rendered": "Current State is 0",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Increment Count",
            "onPress": [Function],
            "title": "increment",
          },
          "ref": null,
          "rendered": "Increment Count",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
